Fix : 사용하지 않는 className 삭제

Jone4865 opened the pull request:
## 어떤 이유로 PR을 하셨나요?
- [ ] feature 병합
- [ ] 버그 수정
- [X] 코드 개선
- [ ] 기타(아래에 자세한 내용을 기입해주세요)

## 스크린샷 및 세부 내용 - 왜 해당 PR이 필요한지 자세하게 설명해주세요
- build error 해결을 위해 CustomButton 컴포넌트내의 미사용 className 변수를 제거하였습니다.

heum2 committed:
Design: 레이아웃 UI 변경

Jone4865 committed:
Merge branch 'develop' of github.com:CarpoolMATE/FRONTEND into design/jongone/sign-up

Jone4865 committed:
feat : 공통사용 color 추가(primary, error), 공통 버튼 구현

Jone4865 committed:
Fix : build 오류 수정

heum2 commented on src/components/button.tsx:
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  intent?: 'default' | 'delete' | 'out_line';
```
P3: `intent`의 옵션 중에 `out_line`으로 되어있는데, `outline` 은 어떤가요?

Jone4865 replied:
intent - out_line > outline으로 변경

heum2 commented on src/components/button.tsx:
```diff
@@ -0,0 +1,31 @@
+}) => {
+  return (
+    <button
+      className={`font-pretendard cursor-pointer w-full rounded-xl font-semibold text-lg hover:opacity-95 active:opacity-95 disabled:border-none disabled:bg-[#DADDE1] disabled:cursor-not-allowed disabled:text-[#A2ABB4] disabled:hover:opacity-100
+        ${size === 'default' ? 'h-[51px]' : size === 'small' ? 'h-11' : ''}
+        ${
+          intent === 'default'
+            ? 'bg-primary text-white'
+            : intent === 'delete'
+              ? 'bg-error text-white'
+              : 'border-[1px] border-primary text-primary bg-white hover:bg-[#F2F8FF]'
+        }`}
```
P2: `className`이 하나에 너무 많이 들어가 있는 것 같습니다. `classnames` 라이브러리를 설치해서 조건에 맞게 분리하여 합쳐주는 건 어떤가요? 
또한, 버튼의 확장성을 위해 `className` 도 포함시켜주면 좋을 것 같습니다.

```tsx
import classNames from 'classnames';

const sizeClasses = {
  'h-[51px]': size === 'default',
  'h-11': size === 'small',
};

return (
  <button className={classNames(sizeClasses, props.className)} {...props}> 
    {props.children}
  </button>
)
```

Jone4865 replied:
classnames 설치 및 className 확장 적용

heum2 commented on src/components/button.tsx:
```diff
@@ -0,0 +1,31 @@
+  size?: 'default' | 'small';
+};
+
+const CustomButton: React.FC<ButtonProps> = ({
+  intent = 'default',
+  size = 'default',
+  ...props
+}) => {
```
P1: 컴포넌트에 `React.FC` 를 다 붙일 필요 없을 것 같습니다. 아래와 같이 변경해주세요.

```suggestion
const CustomButton = ({
  intent = 'default',
  size = 'default',
  ...props
}: ButtonProps) => {
```

Jone4865 replied:
React.FC 제거

heum2 commented:
리뷰를 작성할 때, `label` 과 해당되는 체크박스에 해주세요!

Jone4865 committed:
Refactor
1. intent - out_line > outline으로 변경
2. classnames 설치 및 className 확장 적용
3. React.FC 제거
4. 파일명 button > customButton으로 변경

heum2 commented on src/components/customButton.tsx:
```diff
@@ -0,0 +1,31 @@
+  className,
+  ...props
+}: ButtonProps) => {
+  const buttonClasses = classNames(
+    'h-[51px] font-pretendard cursor-pointer w-full rounded-xl font-semibold text-lg hover:opacity-95 active:opacity-95 disabled:border-none disabled:bg-[#DADDE1] disabled:cursor-not-allowed disabled:text-[#A2ABB4] disabled:hover:opacity-100',
+    {
+      'bg-primary text-white': intent === 'default',
+      'bg-error text-white': intent === 'delete',
+      'border border-primary text-primary bg-white hover:bg-[#F2F8FF]':
+        intent === 'outline',
+    },
+    className,
+  );
```
P2: `intent`로 조건 식이 들어가는데, 객체로 만드는건 어떤가요?

예를 들어,
```tsx
const INTENT_CLASSES: Record<Intent, string> = {
  default: 'bg-primary text-white',
  delete: 'bg-error text-white',
  outline: 'border border-primary text-primary bg-white hover:bg-[#F2F8FF]',
};

classNames(..., INTENT_CLASSES[intent], className);
```
위와 같이 한다면 조건 식이 필요 없을 것 같습니다 😊

Jone4865 replied:
intent 객체화 적용 및 조건식 제거

Jone4865 committed:
Refactor
1. intent 객체화 적용 및 조건식 제거

heum2 commented on src/components/customButton.tsx:
```diff
@@ -5,19 +5,20 @@ type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
   intent?: 'default' | 'delete' | 'outline';
 };
 
+const INTENT_CLASSES: Record<'default' | 'delete' | 'outline', string> = {
```
P1: `Intent` 타입을 따로 만들어서 사용해주세요.

```tsx
type Intent = 'default' | 'delete' | 'outline';
```

Jone4865 replied:
상수 및 타입 파일로 분리, 폴더화

heum2 commented on src/components/customButton.tsx:
P1: `type` 들은 `types` 폴더, 상수들은 `constants` 폴더에 정리해주세요.
해당 컴포넌트 내에서만 사용하는 것이니 `Button/types`, `Button/constants`, `Button/index.tsx`  로 정리할 수 있을 것 같습니다.

Jone4865 replied:
상수 및 타입 파일로 분리, 폴더화

heum2 commented on src/components/customButton.tsx:
P3: 이름을 `CustomButton`이 아닌 `Button`으로 해주시면, 자주 사용해야하는 버튼으로 인식 될 것 같은데 어떤가요?

Jone4865 replied:
파일명 customButton > button 수정

Jone4865 committed:
Refactor
1. 파일명 customButton > button 수정
2. 상수 및 타입 파일로 분리

heum2 commented on src/components/button/index.tsx:
```diff
@@ -1,24 +1,16 @@
 import React from 'react';
 import classNames from 'classnames';
+import { ButtonIntentType } from '@/components/button/types';
+import { BUTTON_INTENT_CLASSES } from '@/components/button/constants';
```
P1: `import`를 구분해주세요. 외부라이브러리, `constants`, `types` 분리해야합니다.

heum2 replied:
```tsx
import React from 'react';
import classNames from 'classnames';

import { ButtonIntentType } from '@/components/button/types';

import { BUTTON_INTENT_CLASSES } from '@/components/button/constants';
```

이런식으로 분리해주세요.

Jone4865 replied:
import 구분 추가

heum2 commented:
빌드에러가 발생하는데, `npm i` 한번 해주시고, `package-lock.json` 파일도 같이 올려주세요.

Jone4865 committed:
Refactor
1. import 구분 추가

Jone4865 committed:
Fix : build 에러 수정

Jone4865 committed:
Revert "Fix : build 에러 수정"

This reverts commit 062f0c2864cdfe5caa6a9d251d9da85e4a9ca9ac.

Jone4865 committed:
Fix : 패키지매니저 yarn > npm 변경

Jone4865 commented:
패키지매니저 yarn > npm 변경

heum2 reviewed (approved):
수고하셨습니다. 👍
